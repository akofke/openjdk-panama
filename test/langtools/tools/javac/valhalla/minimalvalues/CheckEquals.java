/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support == or !=
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckEquals.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror  -Xlint:values CheckEquals.java
 */
@jdk.incubator.mvt.ValueCapableClass
final class CheckEquals {
    boolean foo(CheckEquals a, CheckEquals b) {
        return (a == b) || (a != b);
    }
}
