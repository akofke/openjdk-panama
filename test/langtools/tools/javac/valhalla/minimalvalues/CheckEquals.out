- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckEquals.java:10:19: compiler.warn.value.does.not.support: ==
CheckEquals.java:10:31: compiler.warn.value.does.not.support: !=
1 error
3 warnings
